### PR TITLE
added Elastic Scale, crop and Y scale to defish0r

### PR DESCRIPTION
**Pull request by hypov8**
Dynamic Stretch. 
-0.2 to 0.2. 
added a simplified version of the 'Elastic Scale' filter. fixes gopro up-scaled videos 4:3 to 16:9 thats used in GoPro's Superview

Crop. 
true/false. 
crop all 4 borders straight. removing distorted sections

Y scale. 
1.5 to 0.5. 
independently scale Y axis 

added fix for Visual Studio roundf. VS 2013+ supports this function
removed unnecessary calls to roundf in interpNN_b32(nearest neighbor). speed boost

### CODE REVIEW

**Comment by hypov8** [comment]
VS has lots of warning. eg. typed numbers not being float 1.0 instead of 1.0f

conversion from 'double' to 'float', possible loss of data
conversion from 'float' to 'unsigned char', possible loss of data

did you want me to fix any of these issues while i am updating the files?

File: src/filter/defish0r/defish0r.c [hunk 1]
```diff
@@ -345,7 +428,7 @@ void f0r_get_plugin_info(f0r_plugin_info_t* info)
 	info->frei0r_version=FREI0R_MAJOR_VERSION;
 	info->major_version=0;
 	info->minor_version=3;
```
Comment (ddennedy) [inline src/filter/defish0r/defish0r.c:430]: minor_version needs to be incremented

File: src/filter/defish0r/defish0r.c [hunk 2]
```diff
@@ -394,6 +477,21 @@ void f0r_get_param_info(f0r_param_info_t* info, int param_index)
 		info->type = F0R_PARAM_DOUBLE;
 		info->explanation = "Manual Pixel Aspect ratio";
 		break;
+	case 8:
+		info->name = "Crop";
+		info->type = F0R_PARAM_BOOL;
+		info->explanation = "Straighten all edges of video frame";
+		break;
+	case 9:
+		info->name = "Dynamic Stretch";
```
Comment (ddennedy) [inline src/filter/defish0r/defish0r.c:486]: What do you mean by "dynamic" here? Dynamic suggests it is responding to something. If so, what? Can it be simplified as "Stretch"?

File: src/filter/defish0r/defish0r.c [hunk 3]
```diff
@@ -394,6 +477,21 @@ void f0r_get_param_info(f0r_param_info_t* info, int param_index)
...
+		info->type = F0R_PARAM_DOUBLE;
+		info->explanation = "Fix camera dynamic stretch";
+		break;
+	case 10:
+		info->name = "Y Scale";
+		info->type = F0R_PARAM_DOUBLE;
+		info->explanation = "Scale Y to effect aspect ratio";
```
Comment (ddennedy) [inline src/filter/defish0r/defish0r.c:493]: "effect" should be "affect" when used as a verb.

File: src/filter/defish0r/defish0r.c [hunk 4]
```diff
@@ -37,6 +37,39 @@
 
 double PI=3.14159265358979;
 
+//add simplified 'Elestic Scale' to fix superview
+float stretchWidth(int width, int widthCentre, float currentXPos, float stretchFactor)
+{
+	double ratio, lowerWeight = 0, higherWeight = 0, linearRatio;
+	unsigned int lengthSection;
+	float relativeXPos = 0;
```
Comment (ddennedy) [inline src/filter/defish0r/defish0r.c:45]: Add `f` suffix to this float literal.

File: src/filter/defish0r/defish0r.c [hunk 5]
```diff
@@ -37,6 +37,39 @@
...
+		linearRatio = (double) ( currentXPos ) / lengthSection;
+		ratio = sin(linearRatio * PI - PI) * stretchFactor + linearRatio;
+	}
+	else
+	{
+		lengthSection = width - widthCentre - 1;
+		linearRatio = (double) ( currentXPos - widthCentre ) / lengthSection;
+		ratio = sin(linearRatio * PI) * stretchFactor + linearRatio;
+	}
+	ratio = ratio <= 0 ? 0 : ratio;
```
Comment (ddennedy) [inline src/filter/defish0r/defish0r.c:60]: Add `.0` suffix to these numeric literals to make them match the type of `ratio`.

**Review by ddennedy — CHANGES_REQUESTED** [review]

File: src/filter/defish0r/defish0r.c (hunk 2 above)
Comment (hypov8) [reply src/filter/defish0r/defish0r.c:486]: changing to 'Non-Linear scale'

**Review by ddennedy — APPROVED** [review]

**Pull request merged**